wireapp/wire-android
# feat: Display asset upload progress [AR-1671]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1671" title="AR-1671" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />AR-1671</a>  No loading indicator when uploading large file
  </td></table>
  <br />
 




















----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Assets upload was being executed before persisting the meessage into memory and therefore blocking the display of this message until the upload to the server was finishing. Therefore we needed to make some changes in Kalium to add an extra field `assetUploadStatus` that could be persisted and updated accordingly to the final result of the upload api.

### Dependencies (Optional)

- https://github.com/wireapp/kalium/pull/907

Needs releases with:


### Testing

#### Test Coverage (Optional)


#### How to Test

Just open a conversation, select an asset to upload, then double check the asset shows up in the conversation nad goes from upload to success or failure states.

### Attachments (Optional)


https://user-images.githubusercontent.com/2468164/191057696-be11f2bc-9edb-4d59-b130-4f19e4f51d48.mp4


----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
